src/Yaapii.Atoms/Collection/NotEmpty.cs

```diff
@@ -0,0 +1,46 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using Yaapii.Atoms.Error;
+using Yaapii.Atoms.Scalar;
+
+namespace Yaapii.Atoms.Collection
+{
+    /// <summary>
+    /// Ensures that <see cref="ICollection{T}" /> is not empty/>
+    /// </summary>
+    /// <typeparam name="T">Type of the collection</typeparam>
+    public sealed class NotEmpty<T> : CollectionEnvelope<T>
+    {
```

Comment:
class should encapsulate something, in this case original collection and the exception

Reply:
If i create local members in the primary ctor, they will never been used because i have to call the base class ctor of the `CollectionEnvelope<T>` class.

This class already encapsulate all base members.
I have to change the behavior of `base.GetEnumerator()` by calling the base ctor and define a `ScalarOf` in my approach.

To realize your annotation:
The `GetEnumerator()` method is not overrideable.
I can implement `IColllection<T>` directly **or** make `GetEnumerator()` abstract in class `CollectionEnvelope<T>` to be able to override `GetEnumerator()` in the `NotEmpty` class.
Is this your intention?

Reply:
Ok, lets go with your current solution.